hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Client.java

```diff
@@ -733,7 +733,11 @@ public Object run() throws IOException, InterruptedException {
                   + UserGroupInformation.getLoginUser().getUserName() + " to "
                   + remoteId;
               LOG.warn(msg, ex);
-              throw new IOException(msg, ex);
+              throw NetUtils.wrapException(remoteId.getAddress().getHostName(),
+                  remoteId.getAddress().getPort(),
+                  NetUtils.getHostname(),
+                  0,
+                  new IOException(msg, ex));
```

Comment:
+1 for the change.

ex is only ever an IOException; we should preserve it so that the type doesn't get lost in the chain.
* change the signature of the method to be `IOException ex` and then pass it into wrap exception

Reply:
Oh yes, it is already of type IOException, should have seen this earlier and would not have got lost in the stacktrace..
Let me make this change, thanks.

Reply:
> preserve the type of the exception as it goes down. this keeps the information and removes one level of exception wrapping

Agree, updated the PR.